tests/test_integration_layout.py

```diff
@@ -0,0 +1,171 @@
+    (
+        "screen_size",
+        "placeholders_count",
+        "root_container_style",
+        "placeholders_style",
+        "expected_root_widget_virtual_size",
+        "expected_placeholders_size",
+        "expected_placeholders_offset_x",
+    ),
+    (
+        [
+            SCREEN_SIZE,
+            1,
+            "border: ;",  # #root has no border
+            "",  # no specific placeholder style
+            # #root's virtual size=screen size
+            (SCREEN_W, SCREEN_H),
+            # placeholders width=same than screen :: height=default height
+            (SCREEN_W, PLACEHOLDERS_DEFAULT_H),
+            # placeholders should be at offset 0
+            0,
+        ],
+        [
+            # "none" borders still allocate a space for the (invisible) border
+            SCREEN_SIZE,
+            1,
+            "border: none;",  # #root has an invisible border
+            "",  # no specific placeholder style
+            # #root's virtual size is smaller because of its borders
+            (SCREEN_W - 2, SCREEN_H - 2),
+            # placeholders width=same than screen, minus 2 borders :: height=default height minus 2 borders
+            (SCREEN_W - 2, PLACEHOLDERS_DEFAULT_H),
+            # placeholders should be at offset 1 because of #root's border
+            1,
+        ],
+        [
+            SCREEN_SIZE,
+            1,
+            "border: solid white;",  # #root has a visible border
+            "",  # no specific placeholder style
+            # #root's virtual size is smaller because of its borders
+            (SCREEN_W - 2, SCREEN_H - 2),
+            # placeholders width=same than screen, minus 2 borders :: height=default height minus 2 borders
+            (SCREEN_W - 2, PLACEHOLDERS_DEFAULT_H),
+            # placeholders should be at offset 1 because of #root's border
+            1,
+        ],
+        [
+            SCREEN_SIZE,
+            4,
+            "border: solid white;",  # #root has a visible border
+            "",  # no specific placeholder style
+            # #root's virtual height should be as high as its stacked content
+            (SCREEN_W - 2 - 1, PLACEHOLDERS_DEFAULT_H * 4),
+            # placeholders width=same than screen, minus 2 borders, minus scrollbar :: height=default height minus 2 borders
+            (SCREEN_W - 2 - 1, PLACEHOLDERS_DEFAULT_H),
+            # placeholders should be at offset 1 because of #root's border
+            1,
+        ],
+        [
+            SCREEN_SIZE,
+            1,
+            "border: solid white;",  # #root has a visible border
+            "align: center top;",  # placeholders are centered horizontally
+            # #root's virtual size=screen size
+            (SCREEN_W, SCREEN_H),
+            # placeholders width=same than screen, minus 2 borders :: height=default height
+            (SCREEN_W - 2, PLACEHOLDERS_DEFAULT_H),
+            # placeholders should be at offset 1 because of #root's border
+            1,
```

Comment:
before the bugfix we had a offset_x=2 for Placeholders that are horizontally centred, which creates the visual glitch